### DISCUSSION TITLE
supabase.auth.setAuth(access_token) returning null user

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#6913] [Questions]
I'm trying to follow the instructions here: 

https://supabase.com/docs/reference/javascript/auth-setauth

I'm passing the access_token from the client to the server, but when I try to use it in:

`supabase.auth.setAuth(access_token)`

It returns a null user:

```
{
  access_token: [ACCESS TOKEN]
  token_type: 'bearer',
  user: null
}
```

Am I missing something?

**Comment by GaryAustin1** [comment]
That just sets the jwt header basically.  You will need to do that and then get the user info from the server.
Issue report:
[Documentation shows setAuth() returning user and error in the return object. This is causing confusion.](https://github.com/supabase/gotrue-js/issues/247
)

**Reply by jaredcosulich** [reply]
Ok, thank you for clarifying. 

When I try to get the user, though, I get this:

```
{
  user: null,
  data: null,
  error: {
    message: 'Invalid token: token contains an invalid number of segments',
    status: 401
  }
}
```

This is using the access token provided on the client from `supabaseClient.auth.session()`

**Reply by jaredcosulich** [reply]
I figured it out. I was looking for:

```
const { user, error } = await supabaseClient.auth.api.getUser(
    access_token
)
```

This provided the user.
